src/server/twemcache/data/process.c

```diff
@@ -242,23 +245,25 @@ _process_set(struct response *rsp, struct request *req)
     struct item *it;
     struct bstring key;
 
-    status = _put(&istatus, req);
+    status = _put(&istatus, req, true);
```

Comment:
Correct. The reason has much to do with cache invalidation:

For cache, it's generally considered worse to have stale/incorrect data than no data at all. In other words, proactive cache invalidation is desirable for correctness, as long as that doesn't happen on a massive scale (e.g. entire cache invalidated at once). When someone is sending request to update a key, it usually implies that the existing value is no longer valid.

The only exception to this assumption is ```add```, which only wants to insert if the key does not already exist. ```add``` is most commonly used for immutable data in practice. Therefore, failing to write the new value is not ground for deleting the old value, which may very well be the same.

Let me add the reasoning as a note here.